src/Bundle/Maker/MakeGrid.php

```diff
@@ -0,0 +1,285 @@
+    {
+        if (!$input->getArgument('section')) {
+            $question = new ChoiceQuestion(
+                'Please select a section for your grid',
+                ['backend', 'frontend'],
+                0
+            );
+
+            $section = $io->askQuestion($question);
+
+            $input->setArgument('section', $section);
+        }
+
+        if (!$input->getArgument('resource')) {
+            $aliases = $this->resourceHelper->getResourcesAliases();
+            $question = new ChoiceQuestion(
+                'Please select a resource for your grid',
+                array_combine($aliases, $aliases),
+                0
+            );
+            $question->setAutocompleterValues($aliases);
+
+            $resourceAlias = $io->askQuestion($question);
+
+            $input->setArgument('resource', $resourceAlias);
+        }
+
+        if ($resourceAlias = $input->getArgument('resource')) {
+            Assert::true($this->resourceHelper->isResourceAliasExist($resourceAlias), sprintf(
+                    'Resource with alias %s not found',
+                    $resourceAlias
+                )
+            );
+        }
+    }
+
+    public function configureDependencies(DependencyBuilder $dependencies)
+    {
+    }
+
+    public function generate(InputInterface $input, ConsoleStyle $io, Generator $generator)
+    {
+        $fields = [];
+        $isFirstField = true;
+        while (true) {
+            $newField = $this->askForNextField($io, $isFirstField);
+            $isFirstField = false;
+
+            if (null === $newField) {
+                break;
+            }
+
+            $fields = array_merge($fields, $newField);
+        }
+
+        $actions = [
+            'main' => [],
+            'item' => [],
+            'bulk' => [],
+        ];
+
+        foreach ($actions as $section => $data) {
+            $hasSectionAction = $io->confirm(sprintf('Do you have %s actions?', $section), true);
+
+            if ($hasSectionAction) {
+                $newActions = $this->askForNextAction($io, $section);
+                $actions[$section] = array_merge($actions[$section], $newActions);
+            } else {
+                unset($actions[$section]);
+            }
+        }
+
+        $this->generateGridConfigFile($input, $io, $generator, $fields, $actions);
+    }
+
+    private function askForNextField(ConsoleStyle $io, bool $isFirstField)
+    {
+        $io->writeln('');
+
+        if ($isFirstField) {
+            $questionText = 'New field name (press <return> to stop adding fields)';
+        } else {
+            $questionText = 'Add another field? Enter the field name (or press <return> to stop adding fields)';
+        }
+
+        $fieldName = $io->ask($questionText, null, function ($name) {
+            // allow it to be empty
+            if (!$name) {
+                return $name;
+            }
+
+            return $name;
```

Comment:
```suggestion
            return $name ?: null;
```
and removal of if statement above?

Reply:
Good idea.